aspnet/tutorials/dotnet-watch.rst

```diff
@@ -0,0 +1,200 @@
+
+  Hosting environment: Production
+  Content root path: /Users/user/dev/aspnet/Docs/aspnet/tutorials/dotnet-watch/sample/WebApp
+  Now listening on: http://localhost:5000
+  Application started. Press Ctrl+C to shut down.
+
+In a web browser, navigate to ``http://localhost:5000/api/math/sum?a=4&b=5`` and you should see the result ``9``.
+
+If you navigate to ``http://localhost:5000/api/math/product?a=4&b=5`` instead, you'd expect to get the result ``20``. Instead, you get ``9`` again.
+
+We'll fix that.
+
+Adding ``dotnet watch`` to a project
+------------------------------------
+
+1. Add ``Microsoft.DotNet.Watcher.Tools`` to the ``tools`` section of the ``WebApp/project.json`` file as in the example below:
+
+.. literalinclude:: dotnet-watch/sample/WebAppTests/project.json
+   :language: javascript
+   :lines: 21-23
+   :emphasize-lines: 2
+   :dedent: 2
+
+2. Run ``dotnet restore``.
+
+The console output will show messages similar to the ones below:
+
+.. code-block:: bash
+
+  log  : Restoring packages for /Users/user/dev/aspnet/Docs/aspnet/tutorials/dotnet-watch/sample/WebApp/project.json...
+  log  : Restoring packages for tool 'Microsoft.DotNet.Watcher.Tools' in /Users/user/dev/aspnet/Docs/aspnet/tutorials/dotnet-watch/sample/WebApp/project.json...
+  log  : Installing Microsoft.DotNet.Watcher.Core 1.0.0-preview2-final.
+  log  : Installing Microsoft.DotNet.Watcher.Tools 1.0.0-preview2-final.
+
+Running ``dotnet`` commands using ``dotnet watch``
+--------------------------------------------------
+
+Any ``dotnet`` command can be run with  ``dotnet watch``:  For example:
+
+========================================= ======================================
+Command                                   Command with watch
+========================================= ======================================
+``dotnet run``                            ``dotnet watch run``
+``dotnet run --f net451``                 ``dotnet watch run --f net451``
+``dotnet run --f net451 -- -arg1``        ``dotnet watch run -f net451 -- -arg1``
+``dotnet test``                           ``dotnet watch test``
+========================================= ======================================
+
+To run ``WebApp`` using the watcher, run ``dotnet watch run`` in the ``WebApp`` folder. The console output will show messages similar to the ones below, indicating that ``dotnet watch`` is now watching code files:
+
+.. code-block:: bash
+
+  user$ dotnet watch run
+  [DotNetWatcher] info: Running dotnet with the following arguments: run
+  [DotNetWatcher] info: dotnet process id: 39746
+  Project WebApp (.NETCoreApp,Version=v1.0) was previously compiled. Skipping compilation.
+  Hosting environment: Production
+  Content root path: /Users/user/dev/aspnet/Docs/aspnet/tutorials/dotnet-watch/sample/WebApp
+  Now listening on: http://localhost:5000
+  Application started. Press Ctrl+C to shut down.
+
+Making changes with ``dotnet watch``
+------------------------------------
```

Comment:
Here you are starting another section, you should say something about "make sure you have dotnet watch running on the WebApp"